.github/workflows/main.yml

```diff
@@ -0,0 +1,47 @@
+# Adapted form https://github.com/avsm/hello-world-action-ocaml/blob/master/.github/workflows/workflow.yml
+name: Main workflow
+
+on:
+  - pull_request
+
+jobs:
+  build:
+    strategy:
+      fail-fast: false
+      matrix:
+        os:
+          # - macos-latest
+          - ubuntu-latest
+          # - windows-latest
+        ocaml-version:
+          - 4.11.0
+          # - 4.10.1
+          # - 4.09.1
+          # - 4.08.1
+
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v2
+
+      - name: Cache opam repository
+        uses: actions/cache@v2
+        with:
+          path: ~/.opam
+          key: ${{ runner.os }}-${{ hashFiles('*.opam') }}-${{ matrix.ocaml-version }}
+
+      - name: Use OCaml ${{ matrix.ocaml-version }}
+        uses: actions-ml/setup-ocaml@master
+        with:
+          ocaml-version: ${{ matrix.ocaml-version }}
+
+      - run: opam pin add emojitsu.dev . --no-action
+
+      - run: opam depext emojitsu --yes --with-doc --with-test
```

Comment:
These steps are now running by default, so you can remove them.

Reply:
Oh nice! Thanks for taking a look and for the tips! :)

Reply:
It can also be disabled via input if necessary: https://github.com/actions-ml/setup-ocaml/blob/master/action.yml
But I'm thinking of allowing glob to determine which local packages to pin, so this API could change in the future.

Reply:
Also note that at this moment you cannot speed up the opam install process unless you disable sandboxing: https://github.com/ocaml/dune/issues/4166

Reply:
Excellent! Thanks :)

I just noticed the inputs, and enabled the dune cache. Seem to be working well so far.

I'm well aware I'm playing with WIP and in-development infra here (this is just a toy project, mainly to play with infra and releases etc.), so I won't get overly attached to the current API :)

After I added caching of the .opam directory, my entire build cycle is under 1:30. :clap:

Reply:
Great. I sometimes do something similar :)

Reply:
You need to disable sandboxing when you use dune-cache. (At least until opam 2.1.0 is released)

Reply:
Noted! Thanks!